ansible/roles/polkadot-validator/templates/proxy.conf.j2

```diff
@@ -0,0 +1,4 @@
+server {
+  listen 0.0.0.0:{{ proxy_port }};
+  proxy_pass localhost:{{ p2p_port }};
+}
```

Comment:
@fgimenez  Why listen on port 80 and forward to 30333? Shouldn't this be reversed?

Same in the Polkadot service config:

```j2
  --public-addr=/ip4/{{ hostvars[inventory_hostname].public_ip.json.ip }}/tcp/{{ proxy_port }} \
  --listen-addr=/ip4/127.0.0.1/tcp/{{ p2p_port }} \
```

Ideally this should be reversed in order to keep the 30333 port, but honestly I'm not sure if changing default ports can actually cause any weird behavior in communication (probably @mxinden can clarify).

I guess in the end it's just up to preferences.

Reply:
@lamafab The port open to the outside network is 80, and the node is listening in 30333, which is not accessible from outside. The proxy listens on 80 and forwards to 30333. The node is instructed to advertise itself using port 80, so that other nodes try to connect to it through the proxy. Does that make sense?

Reply:
>  Does that make sense?

Yes, but I'm just wondering why those port numbers aren't reversed: the publicly accessible port is 30333, and the node is listening locally on 80.

Reply:
Otherwise everything looks OK.

Reply:
> Yes, but I'm just wondering why those port numbers aren't reversed: the publicly accessible port is 30333, and the node is listening locally on 80.

By default the node listens locally on 30333, why would we need to change that?. Using port 80 as the publicly accessible port makes the node reachable by nodes running behind restrictive firewalls, but i agree that this could be any other port number.

Reply:
Also, setting the node to listen on a port < 1024 would require to elevate the privileges of the polkadot user.